conformance/lang/expressions/mapping_constructor_expr.balt

```diff
@@ -0,0 +1,2077 @@
+    any var3 = c;
+    io:println(var3 is record {| int field1; string field2; boolean field3?; |}); // @output true
+
+    record {|int field1; string field2; boolean field3?;|} d = {field1: 0, field2: "mouse"};
+    any var4 = d;
+    io:println(var4 is record {|int field1; string field2; boolean field3?;|}); // @output true
+
+    record {|int field1; string field2; boolean...; |} e = {field1: 23, field2: "Car", "field3": false};
+    any var5 = e;
+    io:println(var5 is record {|int field1; string field2; boolean...; |}); // @output true
+
+    record {|int field1; string field2; boolean...; |} f = {field1: 23, field2: "Car"};
+    any var6 = f;
+    io:println(var6 is record {|int field1; string field2; boolean...; |}); // @output true
+
+    record {|int field1; *MyRecord;|} g = {field1: 43, myField1: 3443, myField2: false};
+    any var7 = g;
+    io:println(var7 is record {| int field1; int myField1; boolean myField2; |}); // @output true
+
+    MyRecord h = {myField1: 776777777, myField2: true};
+    any var8 = h;
+    io:println(var8 is MyRecord); // @output true
+
+    record {int field1; string field2; boolean field3;} i = {field1: 0, field2: "mouse", field3: false};
+    any var9 = i;
+    io:println(var9 is record {| int field1; string field2; boolean field3; anydata...; |}); // @output true
+
+    record {int field1; string field2;} j = {field1: 0, field2: "mouse", "field3": false};
+    any var10 = j;
+    io:println(var10 is record {| int field1; string field2; anydata...; |}); // @output true
+
+    record {int field1; string field2; boolean field3?;} k = {field1: 6756, field2: "cat", field3: false};
+    any var11 = k;
+    io:println(var11 is record {| int field1; string field2; boolean field3?; anydata...; |}); // @output true
+
+    record {int field1; string field2; boolean field3?;} l = {field1: 6756, field2: "cat", "other": "false"};
+    any var12 = l;
+    io:println(var12 is record {| int field1; string field2; boolean field3?; anydata...; |}); // @output true
+
+    record {int field1; *MyRecord;} m = {field1: 43, myField1: 3443, myField2: false};
+    any var13 = m;
+    io:println(var13 is record {| int field1; int myField1; boolean myField2; anydata...; |}); // @output true
+
+    record {int field1; *MyRecord;} n = {field1: 43, myField1: 3443, myField2: false, "other": 43434};
+    any var14 = n;
+    io:println(var14 is record {| int field1; int myField1; boolean myField2; anydata...; |}); // @output true
+
+    MyRecord2 o = {myField1: 776777777, myField2: true, "other": false};
+    any var15 = o;
+    io:println(var15 is MyRecord2); // @output true
+
+    json p = {myField1: 776777777, myField2: true, "other": false};
+    any var16 = p;
+    io:println(var16 is map<json>); // @output true
+
+    any q = {myField1: 776777777, myField2: true, "other": false};
+    any var17 = q;
+    io:println(var17 is  map<any>); // @output true
```

Comment:
This is false. See https://github.com/ballerina-platform/ballerina-lang/issues/33377.